src/server/conn.rs

```diff
@@ -564,6 +568,18 @@ where
         }
     }
 
+    /// Prevent shutdown of the underlying IO object at the end of service the request,
+    /// instead run `into_parts`. This is a convenience wrapper over `poll_without_shutdown`.
+    pub fn without_shutdown(self) -> impl Future<Item=Parts<I,S>, Error=::Error>
+            where S:'static, I:'static
```

Comment:
Are these `'static` bounds required? I wouldn't think so, since they aren't needed to call `poll_without_shutdown` nor `into_parts`...

Reply:
Looks like "no"... I was confused by `'static` bounds in `Connection`'s `Future` implementation.

They leaked into disappeared type `ConnectionNoShutdow` and before the refactoring it failed to compile without such bounds.